examples/Stratigraphic_model/serial.py

```diff
@@ -0,0 +1,68 @@
+from firedrake import *
+from firedrake.petsc import PETSc
+print = lambda x: PETSc.Sys.Print(x)
+
+
+class Problem(object):
+    def __init__(self, N, degree):
+        super().__init__()
+        self.N = N
+        self.degree = degree
+
+    def mesh(self):
+        mesh = RectangleMesh(self.N, self.N, 100, 100, quadrilateral=False)
+        return mesh
+
+    def initial_condition(self, Z):
+        (x, y) = SpatialCoordinate(Z.mesh())
+        s = Function(Z)
+        s.interpolate(Constant(0))
+        return s
+
+    def function_space(self, mesh):
+        Ve = FiniteElement("CG", mesh.ufl_cell(), self.degree)
+        return FunctionSpace(mesh, Ve)
+
+
+if __name__ == "__main__":
+    N = 1000
+    problem = Problem(N, 1)
+    mesh = problem.mesh()
+    Z = problem.function_space(mesh)
+    s0 = problem.initial_condition(Z)
+    s = Function(Z)
+    q = TestFunction(Z)
+    x, y = SpatialCoordinate(mesh)
+    print("Z.dim():%s" % Z.dim())
+
+    sp = {
+        "snes_max_it": 100,
+        "snes_atol": 1.0e-8,
+        "snes_rtol": 1.0e-8,
+        "snes_monitor": None,
+        "snes_converged_reason": None,
+        "mat_type": "aij",
+        "ksp_type": "preonly",
+        "pc_type": "lu",
+        "pc_factor_mat_solver_type": "mumps",
+    }
+
+    dt = Constant(1000)
```

Comment:
Please can you add some comments to this section to say what the coefficients and forms are?